core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -744,7 +732,7 @@ private[spark] class BlockManager(
       level: StorageLevel,
       tellMaster: Boolean = true,
       effectiveStorageLevel: Option[StorageLevel] = None,
-      keepReadLock: Boolean = false): Option[PutResult] = {
+      keepReadLock: Boolean = false): Option[Option[Iterator[Any]]] = {
```

Comment:
Why do levels of options? What does it mean if the outer is set but the inner is not